packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1635,15 +1635,16 @@ export class TeamsAppSolution implements Solution {
 
       if (ctx.answers?.platform === Platform.CLI) {
         const aadAppTenantId = ctx.envInfo.profile?.get(PluginNames.AAD)?.get(REMOTE_TENANT_ID);
+        const envName = ctx.envInfo.envName ?? "dev";
```

Comment:
If envInfo.envName is undefined, the Toolkit is in an invalid state. I think it would be better to just fail.

Reply:
Will this happen when calling into solution? Or shall we check at the beginning of every stage?

Reply:
In theory, envName will not be undefined. But IMHO, checking the null-ness of envName is good defensive programming practice.

Reply:
Fixed. Thanks!